response_operations_ui/views/accounts.py

```diff
@@ -40,6 +49,58 @@ def get_my_account():
         return jsonify(e)
 
 
+@account_bp.route("/my-account", methods=["POST"])
+def update_account():
+    form = OptionsForm()
+    form_valid = form.validate()
+    if not form_valid:
+        flash("You need to choose an option")
+        return redirect(url_for("account_bp.get_my_account"))
+    else:
+        return redirect(url_for(form_redirect_mapper.get(form.data["option"])))
+
+
+@account_bp.route("/change-account-name", methods=["GET", "POST"])
+def change_account_name():
+    try:
+        form = ChangeAccountName()
+        user_id = session["user_id"]
+        user_from_uaa = uaa_controller.get_user_by_id(user_id)
+        user = {
+            "first_name": f"{user_from_uaa['name']['givenName']}",
+            "last_name": f"{user_from_uaa['name']['familyName']}",
+        }
+        payload = user_from_uaa
+        payload["name"] = {"familyName": form.data["last_name"], "givenName": form.data["first_name"]}
+        if request.method == "POST" and form.validate():
+            if (form.data["first_name"] != user["first_name"]) or (form.data["last_name"] != user["last_name"]):
+                errors = uaa_controller.update_user_account(payload)
+                if errors is None:
+                    full_name = f"{form.data['first_name']} {form.data['last_name']}"
+                    logger.info("Sending update account details email", user_id=user_id)
+                    personalisation = {
+                        "first_name": user["first_name"],
+                        "value_name": "name",
+                        "changed_value": full_name,
+                    }
+                    try:
+                        NotifyController().request_to_notify(
+                            email=user_from_uaa["emails"][0]["value"],
+                            template_name="change_account_details",
+                            personalisation=personalisation,
+                        )
+                        return logout.logout("Your name has been changed")
+                    except NotifyError as e:
+                        logger.error(
+                            "Error sending change of name acknowledgement email to Notify Gateway", msg=e.description
+                        )
+            else:
+                return redirect(url_for("account_bp.get_my_account"))
+        return render_template("account/change-account-name.html", user=user, form=form, errors=form.errors)
+    except Exception as e:
```

Comment:
Don't do this in a view.  You'd go from a styled html page to rendering a blob of json which would be suuuuper jarring for whoever is using the site.

Remove the try/except from this.  If an exception is thrown that isn't catched, it'll bubble up and eventually get caught by errorhandlers (that are found in view/errors.py) and depending on the type of error it'll be handled in a certain way.